src/Extensions/Cassandra.AppMetrics/Cassandra.AppMetrics.csproj

```diff
@@ -0,0 +1,36 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <Description>This package builds on the DataStax Enterprise C# driver and DataStax C# Driver for Apache Cassandra, adding a metrics provider implementation using App Metrics.</Description>
+    <Copyright>Copyright © by DataStax</Copyright>
+    <AssemblyVersion>1.0.0.0</AssemblyVersion>
+    <FileVersion>1.0.0.0</FileVersion>
+    <VersionPrefix>1.0.0</VersionPrefix>
+    <Authors>DataStax</Authors>
+    <TargetFrameworks Condition="'$(BuildCoreOnly)' != 'True'">netstandard2.0;net461</TargetFrameworks>
+    <TargetFrameworks Condition="'$(BuildCoreOnly)' == 'True'">netstandard2.0</TargetFrameworks>
+    <NoWarn>$(NoWarn);1591</NoWarn>
+    <Authors>DataStax</Authors>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <AssemblyName>Cassandra.AppMetrics</AssemblyName>
+    <Product>DataStax C# Drivers App Metrics Extension</Product>
+    <AssemblyOriginatorKeyFile>../../../build/datastax.snk</AssemblyOriginatorKeyFile>
+    <SignAssembly>true</SignAssembly>
+    <PackageId>CassandraCSharpDriver.AppMetrics</PackageId>
+    <Title>DataStax C# Drivers App Metrics Extension</Title>
+    <PackageTags>cassandra;apache;datastax;dse;driver;client;metrics;appmetricsdatabase;nosql;dotnet;netcore;db</PackageTags>
+    <PackageLicenseUrl>https://github.com/datastax/csharp-driver/blob/master/LICENSE</PackageLicenseUrl>
+    <RepositoryUrl>https://github.com/datastax/csharp-driver</RepositoryUrl>
+    <PackageProjectUrl>https://github.com/datastax/csharp-driver</PackageProjectUrl>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="App.Metrics.Abstractions" Version="[3.0.0,4.0)" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\Cassandra\Cassandra.csproj" />
```

Comment:
Here is the generated nuspec inside the nupkg returned by `dotnet pack Cassandra.AppMetrics.csproj`:

```
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2013/05/nuspec.xsd">
  <metadata>
    <id>CassandraCSharpDriver.AppMetrics</id>
    <version>1.0.0-alpha2</version>
    <title>DataStax C# Drivers App Metrics Extension</title>
    <authors>DataStax</authors>
    <owners>DataStax</owners>
    <requireLicenseAcceptance>false</requireLicenseAcceptance>
    <licenseUrl>https://github.com/datastax/csharp-driver/blob/master/LICENSE</licenseUrl>
    <projectUrl>https://github.com/datastax/csharp-driver</projectUrl>
    <description>This package builds on the DataStax Enterprise C# driver and DataStax C# Driver for Apache Cassandra, adding a metrics provider implementation using App Metrics.</description>
    <copyright>Copyright © by DataStax</copyright>
    <tags>cassandra apache datastax dse driver client metrics appmetricsdatabase nosql dotnet netcore db</tags>
    <repository url="https://github.com/datastax/csharp-driver" />
    <dependencies>
      <group targetFramework=".NETFramework4.6.1">
        <dependency id="CassandraCSharpDriver" version="3.11.0" exclude="Build,Analyzers" />
        <dependency id="App.Metrics.Abstractions" version="3.1.0" exclude="Build,Analyzers" />
      </group>
      <group targetFramework=".NETStandard2.0">
        <dependency id="CassandraCSharpDriver" version="3.11.0" exclude="Build,Analyzers" />
        <dependency id="App.Metrics.Abstractions" version="3.1.0" exclude="Build,Analyzers" />
      </group>
    </dependencies>
  </metadata>
</package>
```

It looks like it is correctly picking up the dependency as a package dependency. If I'm not mistaken this was one of the features of the new `dotnet` CLI when .NET Core came out, i.e., treating every project as a package.

Reply:
Would it be possible to use version ranges?

It would be nice if we could support multiple `CassandraCSharpDriver` versions from a single `CassandraCSharpDriver.AppMetrics` version, avoiding to publish one `CassandraCSharpDriver.AppMetrics` version per `CassandraCSharpDriver` version.

Is it possible to support version ranges of `CassandraCSharpDriver`?

Reply:
So by default `CassandraCSharpDriver.AppMetrics` will require `CassandraCSharpDriver` >= 3.11.0 assuming 3.11.0 is the version of `CassandraCSharpDriver`  at the time when `dotnet pack` is run for `CassandraCSharpDriver.AppMetrics`. With this we wouldn't have to build `CassandraCSharpDriver.AppMetrics` everytime but it does mean that whenever we build `CassandraCSharpDriver.AppMetrics` for a new release we are bumping the dependency version requirement. 

Are you talking about manually setting the version requirement like  `CassandraCSharpDriver` >= 3.0 and < 4.0? This is doable but with a manually created `.nuspec` only (the old way)